proto-lens-protoc/src/Data/ProtoLens/Compiler/Definitions.hs

```diff
@@ -191,6 +224,13 @@ definedFieldType fd env = fromMaybe err $ Map.lookup (fd ^. typeName) env
     err = error $ "definedFieldType: Field type " ++ unpack (fd ^. typeName)
                   ++ " not found in environment."
 
+-- | Look up the Haskell name for the type of a given type.
+definedType :: Text -> Env QName -> Definition QName
```

Comment:
Code structure: Do all name resolution within `Definition`, rather than deferring it to `Generate`.  I.e., `ServiceInfo` should contain the Haskell names of the types, not just the proto names, similar to what we do for (sub)messages.  For example, by passing an Env to `collectServices`.

(Also: merge this with the function at l.221)

Reply:
How strongly do you feel about this? `hsFieldType` and friends already do this in `Generate`, and there is no unified `Env` to lookup until `Generate.`

Reply:
You're right, my bad.  The way you wrote it SGTM.